src/plugins/workspace/public/containers/workspace_dropdown_list/workspace_dropdown_list.tsx

```diff
@@ -50,17 +49,19 @@ export function WorkspaceDropdownList(props: WorkspaceDropdownListProps) {
     [allWorkspaceOptions]
   );
 
-  const onChange = (workspaceOption: WorkspaceOption[]) => {
-    /** switch the workspace */
-    setLoading(true);
-    onSwitchWorkspace(workspaceOption[0].key!)
-      .catch((err) =>
-        coreStart.notifications.toasts.addDanger('some error happens in workspace service')
-      )
-      .finally(() => {
-        setLoading(false);
-      });
-  };
+  const onChange = useCallback(
+    (workspaceOption: WorkspaceOption[]) => {
+      /** switch the workspace */
+      setLoading(true);
+      const id = workspaceOption[0].key!;
+      const newUrl = coreStart.workspaces?.formatUrlWithWorkspaceId(window.location.href, id);
+      if (newUrl) {
+        window.location.href = newUrl;
```

Comment:
What about using `navigateToUrl`? Is this API still need get current path?

Reply:
I tried to use `navigateToUrl`, but it seems invalid when calling switch in home.